src/library/scala/collection/Traversable.scala

```diff
@@ -102,6 +102,7 @@ object Traversable extends TraversableFactory[Traversable] { self =>
   implicit def canBuildFrom[A]: CanBuildFrom[Coll, A, Traversable[A]] = ReusableCBF.asInstanceOf[GenericCanBuildFrom[A]]
 
   def newBuilder[A]: Builder[A, Traversable[A]] = immutable.Traversable.newBuilder[A]
+  private[collection] val legacyGroupBy = sys.props.get("scala.groupBy.legacy").map(_.toBoolean).getOrElse(false)
```

Comment:
`java.lang.Boolean.getBoolean("scala.groupBy.legacy")` is simplest and doesn't risk cycles in top-level object initialization.

Reply:
Done

Reply:
It's not implemented in Scala.js, though. Consider directly using `System.getProperty` instead.

Reply:
corrected to use AccessController as that is semantically accurate

Reply:
Er ... That doesn't help. We don't have `AccessController` either. I had never even heard of that one.

So far all the system property-based switches for the behavior of the standard library have been regulated with a simple `System.getProperty` without fuss around it. Can't we keep doing the same?

Reply:
> Er ... That doesn't help. We don't have `AccessController` either. I had never even heard of that one.
> 

https://docs.oracle.com/javase/8/docs/api/java/security/AccessController.html

This is the (only) way to determine if the caller is allowed to perform that SecurityManager managed access
Crudely put, System.getProperty will check if every frame on the call stack is allowed to perform the action
running this via AccessController indicates that the library is taking responsibility for that action, and so we are checking in the scala library is allowed to access properties
This allows SecurityManager to scope permission access appropriately 

> So far all the system property-based switches for the behavior of the standard library have been regulated with a simple `System.getProperty` without fuss around it. Can't we keep doing the same?

This is semantically incorrect (and also therefore incorrect in 3 other places in the library that it looks like it should apply)
(ExecutionContextImpl, and serialization of the RedBlackTrees)

Specifically libraries are what this security feature is all about
I presume that scala.js doesn't support SecurityManager (there is probably no need realistically)

@retronym how do you want to proceed?

Reply:
Sorry for sidestepping the discussion, but do we really need this PR? We should be free to change the iteration order of maps in minor releases.

Reply:
> It's not implemented in Scala.js, though. Consider directly using `System.getProperty` instead.

`java.lang.getBoolean` is used elsewhere. Came across https://github.com/scala/scala/blob/2.12.x/src/library/scala/reflect/ClassTag.scala#L140

Reply:
I had not noticed that one. We have a completely different implementation of `ClassTag` in Scala.js, with a separate copy of the file that maintains the binary API.